src/test/ui/fn/fn-item-type.rs

```diff
@@ -1,53 +1,52 @@
 // Test that the types of distinct fn items are not compatible by
 // default. See also `run-pass/fn-item-type-*.rs`.
 
-fn foo<T>(x: isize) -> isize { x * 2 }
-fn bar<T>(x: isize) -> isize { x * 4 }
+fn foo<T>(x: isize) -> isize {
+    x * 2
+}
+fn bar<T>(x: isize) -> isize {
+    x * 4
+}
 
-fn eq<T>(x: T, y: T) { }
+fn eq<T>(x: T, y: T) {}
 
-trait Foo { fn foo() { /* this is a default fn */ } }
-impl<T> Foo for T { /* `foo` is still default here */ }
+trait Foo {
+    fn foo() { /* this is a default fn */
+    }
+}
+impl<T> Foo for T {
+    /* `foo` is still default here */
+}
 
 fn main() {
     eq(foo::<u8>, bar::<u8>);
     //~^ ERROR mismatched types
     //~| expected fn item `fn(_) -> _ {foo::<u8>}`
     //~| found fn item `fn(_) -> _ {bar::<u8>}`
     //~| expected fn item, found a different fn item
-    //~| different `fn` items always have unique types, even if their signatures are the same
-    //~| change the expected type to be function pointer
-    //~| if the expected type is due to type inference, cast the expected `fn` to a function pointer
```

Comment:
Where did these notes go? Is there a good reason they disappeared?

Reply:
The note about fn items being unique is still there, however the rest of the deleted notes are making a suggestion that would not result in working code. 

If you follow the advice of casting to function pointers (in this example), it will require having `std::ptr::eq` in scope. 

If you follow that advice, this is the result. 

```
error[E0308]: arguments to this function are incorrect
    --> cases/1.rs:19:5
     |
19   |     eq(foo::<u8> as fn(isize) -> isize, bar::<u8> as fn(isize) -> isize);
     |     ^^
     |
note: expected `*const _`, found fn pointer
    --> cases/1.rs:19:8
     |
19   |     eq(foo::<u8> as fn(isize) -> isize, bar::<u8> as fn(isize) -> isize);
     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     = note: expected raw pointer `*const _`
                 found fn pointer `fn(isize) -> isize`
note: expected `*const _`, found fn pointer
    --> cases/1.rs:19:41
     |
19   |     eq(foo::<u8> as fn(isize) -> isize, bar::<u8> as fn(isize) -> isize);
     |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     = note: expected raw pointer `*const _`
                 found fn pointer `fn(isize) -> isize`
note: function defined here
    --> /home/mperez/projects/rustlang/library/core/src/ptr/mod.rs:1827:8
     |
1827 | pub fn eq<T: ?Sized>(a: *const T, b: *const T) -> bool {
     |        ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
```

So while the note might be useful for some cases, it wouldn't be a good suggestion in this case.

Reply:
I'll come up with a couple other similar cases real quick to see if it might be a good/useful suggestion there.

Reply:
Sorry, I'm still having a hard time understanding why this suggestion is incorrect for this UI test. Why are you pointing at `std::ptr::eq` when the UI test defines its own `fn eq<T>(x: T, y: T) {}`?

For the record, this code works:

```rust
fn eq<T: Eq>(a: T, b: T) {}

fn a() {}
fn b() {}

fn main() {
    eq(a as fn(), b as fn());
    //   ^^^^^^^    ^^^^^^^
}
```

Reply:
You were right. I mistakenly didn't include the locally defined `fn eq` when verifying the output.

Reply:
So can we bring this suggestion back then?